### ISSUE TITLE
Document behaviour of formatting methods on invalid DateTimes

### ISSUE THREAD

**Issue by ab-pm** [moment/luxon#1474]
Like others (#1421) I was caught off-guard by the [typings update](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64995) that declared many return values to be nullable. I had read the [validity docs](https://moment.github.io/luxon/#/validity) when I first encountered the library, but have forgotten them since (albeit using `isValid` in a few places) and not thought about what this meant for formatting methods like `.toString()` or `.toISO()`.

The behaviour of returning `null` on invalid `DateTime` objects is completely missing from the [API documentation](https://moment.github.io/luxon/api-docs/index.html) afaics.

The TypeScript typings should not be the only reminder about this. I would expect the return type of all the `to…` methods to include `| null` in the API docs, a sentence like "*If invalid, returns `null`.*" added to each of their descriptions, and also a section (warning) about this behaviour on the [formatting docs](https://moment.github.io/luxon/#/formatting).

**Comment by icambron** [comment]
Sure would take PRs adding these

**Comment by fantasy0v0** [comment]
I can't quite grasp this. Why not make the fromISO method throw an error or return null directly when it encounters invalid content? Why bother checking for validity in each method instead?

**Comment by diesieben07** [comment]
Absolutely, @fantasy0v0. My plan for the next major version of Luxon includes getting rid of the concept of "invalid date times". `fromISO` would throw immediately.

You can get this behavior today by setting `Settings.throwOnInvalid = true`.

**Comment by dvins** [comment]
But `throwOnInvalid` also desperately needs to be an optional attribute of DateTimeOptions, so you can say `const dt = DateTime.fromISO('some value', { throwOnInvalid: true })`.
